packages/iov-ethereum/src/abi.ts

```diff
@@ -112,6 +117,35 @@ export class Abi {
     return data.slice(32, 32 + length);
   }
 
+  public static decodeSwapProcessState(data: Uint8Array): SwapProcessState {
```

Comment:
The `Abi` class should not contain helpers for a specific kind of contract. But this we can refactor later.

Reply:
Why not? The only reason we deal with ABI is to interact with a specific kind of contract.

Reply:
Yes, but I think it would be good to have explicit dependencies, like:

`Abi`: lowe level ABI helpers for all smart contracts we support
`AtomicSwapAbi` (depends on `Abi`): ABI helpers specific to our atomic swap contracts
`Erc20Abi` (depends on `Abi`): ABI helpers specific all ERC20 contracts

This way you know that something is wrong when you import `AtomicSwapAbi` in ERC20 code. Also you prevent strange dependencies inside of `Abi`.

Reply:
OK good, then I'll leave it for now and we can refactor.